hypervisor/debug/uart16550.c

```diff
@@ -155,7 +155,9 @@ void uart16550_init(bool early_boot)
 				uart.reg_width = 1;
 				pci_pdev_write_cfg(uart.bdf, PCIR_COMMAND, 2U, cmd | PCIM_CMD_PORTEN);
 			} else {
-				if ((bar0 & 0xfU) == 0U) { /* 32 bits MMIO Space */
+				uint32_t bar_hi = pci_pdev_read_cfg(uart.bdf, pci_bar_offset(1), 4U);
+
+				if (((bar0 & 0xfU) == 0U) || (((bar0 & 0xfU) == 4U) && (bar_hi == 0U))) {
```

Comment:
do we really need consider the whole low 4 bits, should we just consider 3 bits (remove bit for prefetchable)?

Reply:
+1. No need to checkk bit 3.

Reply:
BTW, why we cannot support pure 64 bits MMIO serial card?

Reply:
@dongyaozu This is because we didn't do memory remap with 4G+ memory space in uart initialization. I will add a "TODO" in updated version and implement in the future. This PR is a quick fix for the latest EHL SBL update.

Reply:
Need to add comments in this place too. Otherwise, people will be confused why we do so.